runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/HttpBody.kt

```diff
@@ -190,21 +199,25 @@ public suspend fun HttpBody.readAll(): ByteArray? = when (this) {
 /**
  * Convert an [HttpBody] variant to the corresponding [ByteStream] variant or null if empty.
  */
-public fun HttpBody.toByteStream(): ByteStream? = when (val body = this) {
-    is HttpBody.Empty -> null
-    is HttpBody.Bytes -> object : ByteStream.Buffer() {
-        override val contentLength: Long? = body.contentLength
-        override fun bytes(): ByteArray = body.bytes()
-    }
-    is HttpBody.ChannelContent -> object : ByteStream.ChannelStream() {
-        override val contentLength: Long? = body.contentLength
-        override val isOneShot: Boolean = body.isOneShot
-        override fun readFrom(): SdkByteReadChannel = body.readFrom()
-    }
-    is HttpBody.SourceContent -> object : ByteStream.SourceStream() {
-        override val contentLength: Long? = body.contentLength
-        override val isOneShot: Boolean = body.isOneShot
-        override fun readFrom(): SdkSource = body.readFrom()
+public fun HttpBody.toByteStream(): ByteStream? {
+    if (this is ByteStreamHttpBody) return this.stream // avoid wrapping multiple times
```

Comment:
question: what's the issue with wrapping multiple times?

Reply:
If we already have a `ByteStream` that was converted to `HttpBody` we would end up with

`ByteStream` -> `HttpBody` -> `ByteStream` -> `HttpBody` 

We can avoid all of that by introducing explicit types instead of anonymous objects and check if it's already been wrapped and if so unwrap it instead of wrapping it again.